node/components.go

```diff
@@ -23,6 +25,20 @@ func lightComponents(cfg *Config, store Store) fxutil.Option {
 		baseComponents(cfg, store),
 		fxutil.Provide(services.DASer),
 		fxutil.Provide(services.HeaderExchangeP2P(cfg.Services)),
+		// state components
+		fxutil.ProvideIf(cfg.Core.Remote, state.NewService),
+		fxutil.ProvideIf(cfg.Core.Remote, func(lc fx.Lifecycle) (state.Accessor, error) {
+			ca, err := statecomponents.CoreAccessor(cfg.Services.KeyConf, store.Path(),
+				cfg.Core.RemoteConfig.RemoteAddr)
+			if err != nil {
+				return nil, err
+			}
+			lc.Append(fx.Hook{
+				OnStart: ca.Start,
+				OnStop:  ca.Stop,
+			})
+			return ca, nil
+		}),
```

Comment:
Yeah I can do that. The main issue is -- we wanted full node to be able to *provide* state rather than use a core accessor to access it. I think that having that functionality available in light node for now is good. I can also add it to *bridge* node since it already receives a core endpoint anyway. Wouldn't hurt to implement core_access there as well. Wdyt regarding *full* node?

Reply:
Until we don't implement the FullNode state computation there won't be a way for a full node operator to access the state, considering that we don't support running light and full nodes together(ref to a different ports discussion), so I think it is fine for a full node to rely on a Core as long as this is not a requirement.

Reply:
Resolved https://github.com/celestiaorg/celestia-node/pull/420/commits/46f49d2feec485267a3f995907c9240ef628ef70 ?

Reply:
No. What about Bridge? Don't see a reason for it to be restricted to state access, so let's put this in baseComponents.
> Any node should be able to access the state

Reply:
https://github.com/celestiaorg/celestia-node/pull/420/commits/9e3e46bc16cfd28172f5927e1a32856380a27eeb

Now?